### DISCUSSION TITLE
Lyrics support

### DISCUSSION THREAD

**Discussion by potatoeggy** [jellyfin/jellyfin-vue#2224] [Features]
The lyrics API was merged into Jellyfin master a while back and I think it'd be cool to see them in Jellyfin.

What are your thoughts on having it replace the album art when enabled? Kind of like how Musicolet does it:

![image](https://github.com/jellyfin/jellyfin-vue/assets/25178974/70c94a95-1908-4f40-8fde-3959298a82dd)
![image](https://github.com/jellyfin/jellyfin-vue/assets/25178974/58667e66-3980-4ee7-b1a3-6502e3f4489d)

Or, we could push the album art to the left and plop the text directly to its right on devices with larger screens?

Potentially there could be a toggle beside / a menu replacing the current EQ button? This is based on how the visualiser replaces the album art right now. Or maybe it could go beside the playback controls.

![image](https://github.com/jellyfin/jellyfin-vue/assets/25178974/2b373db8-6f03-45b9-adbe-c8ed710e5878)

**Comment by ferferga** [comment]
@potatoeggy For mobile layout I thought about mimicking Spotify, wdyt?

I'm still trying to figure out desktop layout in my head.
Regardless, currently this is very low in my priority queue, first I want to finish all the refactoring work, see [approximate roadmap](https://github.com/orgs/jellyfin/projects/49/views/1)

**Reply by potatoeggy** [reply]
There's already a lyric roadmap item under "short-term" — I'll come back when the refactoring's done, then.

About mimicking Spotify: I'm not a huge fan of Spotify's lyrics UI, which feels bolted onto the existing player as an excuse to use their fun facts cards.
- It reimplements the playback UI but with fewer features in its "full-screen" mode, which is limiting and unnecessary (why not bundle it in the main player instead and reuse its components?)
- There is no way to tell the difference between between a lyric that has already played and a multi-line lyric that is currently playing, which is particularly annoying because its font size is so big
- The mini-lyrics window lets you only see at most six lines (~three lyrics, including the presently playing one since it doesn't occupy the full width, either)
- It's not intuitive if lyrics exist for the song. The "Lyrics" label is always there (at least for songs in Musixmatch's database), requiring you to scroll down to check or recognise that a gray background in the card means that there are lyrics. A button is easier to understand if it's grayed out.
- It looks kinda ugly: the word "Lyrics" one-quarter cut off from the bottom of the screen.
